exec/src/exec.ml

```diff
@@ -51,16 +51,12 @@ let rec exec env evm def =
   Interfaces.force_interface_initializers ();
   let def = Reduction.whd_all env def in
   let (def, args) = app_full def in
-  match kind def with
-  | Construct (c, _)
-    -> (match (Ind.Program.constructor_of c, args) with
-        | (Some Request_program,
-           [instr_t; _ret_t; _instr_ret_t; instr_trm; func_trm])
-          -> let instr_trm = reduce_all env evm instr_trm in
-             exec env evm (exec_request instr_t instr_trm func_trm)
-        | (Some Pure_program, _)
-          -> ()
-        | _
-          -> raise (UnsupportedTerm "coq-exec only execute [Program] terms."))
+  match (Ind.Program.constructor_of def, args) with
+  | (Some Request_program,
+     [instr_t; _ret_t; _instr_ret_t; instr_trm; func_trm])
+    -> let instr_trm = reduce_all env evm instr_trm in
+       exec env evm (exec_request instr_t instr_trm func_trm)
+  | (Some Pure_program, _)
+    -> ()
   | _
-    -> raise (UnsupportedTerm "It was not possible to reduce your term.")
+    -> raise (UnsupportedTerm "coq-exec only execute [Program] terms.")
```

Comment:
Again, we are facing the ambiguity caused by the fact that "Program" is a name already taken in Coq.

I would write `terms of type [FreeSpec.Program]` instead.

Reply:
We can still change the name of the monad, and use something less subject to ambiguity?

`Eff` is a popular name. `Operational` is a bit too long. `program` might be a suitable alternative name, less damaging in terms of “paper-backword compatibility”.

Reply:
Started #21 but I will implement your suggestion nonetheless.